pkg/providers/snow/defaults.go

```diff
@@ -42,12 +46,26 @@ func WithKeyGenerator(generator SshKeyGenerator) DefaultersOpt {
 	}
 }
 
-func (d *Defaulters) GenerateDefaultSshKeys(ctx context.Context, machineConfigs map[string]*v1alpha1.SnowMachineConfig) error {
+// WithUUID will set uuid generated outside of constructor.
+func WithUUID(uuid uuid.UUID) DefaultersOpt {
+	return func(defaulters *Defaulters) {
+		defaulters.uuid = uuid
+	}
+}
+
+// GenerateDefaultSSHKeys generates ssh key if it doesn't exist already.
+func (d *Defaulters) GenerateDefaultSSHKeys(ctx context.Context, machineConfigs map[string]*v1alpha1.SnowMachineConfig, clusterName string) error {
 	md := NewMachineConfigDefaulters(d)
 
 	for _, m := range machineConfigs {
-		if err := md.SetupDefaultSshKey(ctx, m); err != nil {
-			return err
+		if md.keyGenerated {
+			// Generate a random string so that other clusters with same name and devices won't conflict in case user doesn't have access to previous key
+			m.Spec.SshKeyName = md.defaultSSHKeyName(clusterName)
```

Comment:
if a machineconfig has a `sshKeyName`, this line overrides it with a default key which should not be expected behavior

Reply:
I needed to move due to a lint issue that was complaining about cyclic complexity (too many if conditions). Will fix the current condition